annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -105,14 +105,19 @@ public GenomeNexusImpl annotator(String hgvsServiceUrl) {
         this.hgvsServiceUrl = hgvsServiceUrl;
         return this;
     }
+    
+    private boolean annotationNeeded(MutationRecord record) {
+        Map<String, String> additionalProperties = record.getAdditionalProperties();
+        return additionalProperties.containsKey("HGVSp_Short") && additionalProperties.get("HGVSp_Short").isEmpty() && !hgvspNullClassifications.contains(record.getVariant_Classification());
```

Comment:
Sorry to continue to make this complicated, but doesn't this code miss the case where additionalProperties does not contain the key "HGVSp_Short" ?  if it does not, then this method will return false even though annotationNeeded() should be true.

Since we are in a separate method, we don't need to string all these conditions into a single logical expression. Here is my guess as to the logic:

Map<String, String> additionalProperties = record.getAdditionalProperties();
if (!additionalProperties.containsKey("HGVSp_Short") {
return true;
}
return additionalProperties.get("HGVSp_Short").isEmpty() && !hgvspNullClassifications.contains(record.getVariant_Classification());

in other words "if the key was never created, we always have to annotate. If the value of the key is empty we *may* need to annotate, unless the VariantClassification has been set to a non-protein-change type of mutation."

I could be wrong about this ... I think @n1zea144 should weigh in. We can also revert back to the original approach if we have expended too much time on this effort.